src/pipelines/floodscan_pipeline.py

```diff
@@ -0,0 +1,429 @@
+        filename_list = self._get_90_days_filenames_for_dates(dates=dates)
+        zipped_files_path = []
+
+        for zipped_filename in filename_list:
+            sfed_filename = zipped_filename[SFED]
+            mfed_filename = zipped_filename[MFED]
+            sfed_local_file_path = self.local_raw_dir / sfed_filename
+            mfed_local_file_path = self.local_raw_dir / mfed_filename
+
+            if self.mode != "local":
+                try:
+                    download_from_azure(
+                        blob_service_client=self.blob_service_client,
+                        container_name=self.container_name,
+                        blob_path=self.raw_path / sfed_filename,
+                        local_file_path=sfed_local_file_path,
+                    )
+                    download_from_azure(
+                        blob_service_client=self.blob_service_client,
+                        container_name=self.container_name,
+                        blob_path=self.raw_path / mfed_filename,
+                        local_file_path=mfed_local_file_path,
+                    )
+
+                    zipped_files_path.append(
+                        {SFED: sfed_local_file_path, MFED: mfed_local_file_path}
+                    )
+                except Exception as err:
+                    self.logger.error(f"Failed downloading: {err}")
+            else:
+                zipped_files_path.append(
+                    {SFED: sfed_local_file_path, MFED: mfed_local_file_path}
+                )
+
+        return zipped_files_path
+
+    def _unzip_90days_file(self, file_to_unzip, dates):
+        unzipped_files = []
+        try:
+            with ZipFile(file_to_unzip, "r") as zipObj:
+                for fileName in zipObj.namelist():
+                    if os.path.basename(fileName):
+                        date = get_datetime_from_filename(fileName)
+                        if date in dates:
+                            date_str = re.search("([0-9]{4}[0-9]{2}[0-9]{2})", fileName)
+                            new_filename = os.path.basename(
+                                fileName.replace(
+                                    date_str[0], date.strftime(DATE_FORMAT)
+                                )
+                            )
+                            try:
+                                full_path = zipObj.extract(fileName, self.local_raw_dir)
+                                new_full_path = os.path.join(
+                                    os.path.dirname(full_path), new_filename
+                                )
+                                os.rename(full_path, new_full_path)
+                                unzipped_files.append(
+                                    os.path.basename(
+                                        shutil.move(new_full_path, self.local_raw_dir)
+                                    )
+                                )
+                            except Exception:
```

Comment:
This is misleading as many other exceptions might be caught here! Not just the file already existing.